### PR TITLE
Advent of Code 2020 Day 5 solutions

### PR DESCRIPTION
**Pull request by brandonweeks**

### CODE REVIEW

File: src/bin/05.rs [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+            'R' => columns = partition(Half::Upper, &columns).to_vec(),
+            _ => return Err("unexpected control character"),
+        }
+    }
+    Ok(rows[0] * 8 + columns[0])
+}
+
+fn main() -> Result<(), Box<dyn Error>> {
+    let mut seats = Vec::new();
+    for line in fs::read_to_string(INPUT_FILENAME)?.lines() {
```
Comment (twitchyliquid64) [inline src/bin/05.rs:35]: Maybe use `.collect()` and a map function here instead?

File: src/bin/05.rs [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+}
+
+fn partition(half: Half, rows: &[i16]) -> &[i16] {
+    match half {
+        Half::Lower => &rows[..(rows.len() / 2)],
+        Half::Upper => &rows[rows.len() / 2..],
+    }
+}
+
+fn seat_id(input: &str) -> Result<i16, &'static str> {
```
Comment (twitchyliquid64) [inline src/bin/05.rs:18]: What does `i16` represent? Can you make this its own type?

File: src/bin/05.rs (hunk 1 above)
Comment (twitchyliquid64) [reply src/bin/05.rs:35]: https://doc.rust-lang.org/rust-by-example/error/iter_result.html

File: src/bin/05.rs [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    Ok(rows[0] * 8 + columns[0])
+}
+
+fn main() -> Result<(), Box<dyn Error>> {
+    let seats: Result<Vec<_>, _> = fs::read_to_string(INPUT_FILENAME)?
+        .lines()
+        .map(|l| seat_id(l))
+        .collect();
+    let seats = seats?;
+    for seat in 0..seats.len() {
```
Comment (twitchyliquid64) [inline src/bin/05.rs:41]: Btw if you just want to iterate over seats you can do:

```rust
for seat in seats {
  // ye
}
```

Comment (twitchyliquid64) [reply src/bin/05.rs:40]: And if you want to get the index as well you can do:

```rust
for (index, seat) in seats.iter().enumerate() {
  // ye
}
```

`iter()` returns an iterator, and `enumerate()` is implemented for any iterator. `enumerate()` returns a tuple of the index (type `usize`) and the iterator item as an iterator.

Comment (brandonweeks) [reply src/bin/05.rs:40]: `seat in 0..seats.len()` is iterating over all _possible_ seats. One in the range is missing per the puzzle.

Comment (twitchyliquid64) [reply src/bin/05.rs:40]: :+1:

**Pull request merged**